src/test/resources/test-merge-tile/kapt-dinject-tile.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project>
+
+  <description>
+    Add KAPT annotation processor to generate DInject Dependency injection (as java source code).
+  </description>
+
+  <properties>
+    <dinject-generator.version>1.8</dinject-generator.version>
+    <tile-merge-configuration>true</tile-merge-configuration>
```

Comment:
This doesn't feel all that explicit.

Using a magic property seems nasty to begin with, but this seems that merging is dependant on the order in which the tile is injected.

This almost feels like should be explicitly declared at the injection, so something like

    <configuration>
      <tiles>
        <tile merge-with-previous=true>test:test:[123]</tile>
        ...

(again tho - using attributes there is insidious).

Reply:
>  merging is dependant on the order in which the tile is injected.

Well yes in that the `main` target tile should be somewhere before any of the other tiles that add configuration to it but otherwise order of the tiles is moot.

> explicitly declared at the injection

It is the author of the tiles that is explicitly making the choice that these tiles work together. So for me I have been thinking it is the tile author(s) declaring that the tile is a merge target or merge configuration (rather than the tile user).

The user of the tiles just needs to make sure the `main` tile is before the others and for me I was happy to achieve that with documentation (because it's going to be fairly intuitive anyway and if the user gets it wrong it's pretty obvious).  

> <tile merge-with-previous=true>

I think it is more strictly for example ... merge-with="io.avaje.kotlin:kapt".  
This doesn't feel good to me.

> magic property

Happy to put it somewhere else.  It didn't seem to go as an attribute of any of the top level elements (like build) and if I moved it down to configuration ... then we have to traverse down there for every tile when I feel this is potentially the very rare case.

Hmmm.